src/screens/PinScreen.js

```diff
@@ -210,6 +223,8 @@ class PinScreen extends React.Component {
     this.props.setLoadHistoryStatus(this.props.loadHistoryActive, false);
     // show lock screen again
     this.props.lockScreen();
+    // set the wallet state as NOT_STARTED so the LoadingScreen will hide.
+    this.props.startWalletNotStarted();
```

Comment:
Why is it needed? If you don't unlock the wallet, it shouldn't change from state `NOT_STARTED`

Reply:
You are right, we don't need to change it to `NOT_STARTED` anymore as the flow has been refactored

Justed it and the state does not change

Refactored in d89a22dba5b879f471b052db49fe9c6509cfb0ea